sunpy/data/manager/downloader.py

```diff
@@ -0,0 +1,23 @@
+from abc import ABCMeta, abstractmethod
+from parfive import Downloader
+
+
+class DownloaderBase(metaclass=ABCMeta):
+    @abstractmethod
+    def download(self, url, path):
+        """
+        Downloads and returns the path
+        """
+        raise NotImplementedError
+
+
+class ParfiveDownloader(DownloaderBase):
+    def __init__(self):
+        pass
+
+    def download(self, url, path):
+        downloader = Downloader()
+        filename = path.split('/')[-1]
```

Comment:
with Pathlib you could get it as `filename = path.name`

Reply:
I wanted to go Pathlib but not a big fan of `/` :p. Will change everything to use pathlib